d2/src/test/java/com/linkedin/d2/balancer/properties/ClusterPropertiesSerializerTest.java

```diff
@@ -70,30 +70,34 @@ public void testClusterPropertiesSerializer() throws PropertySerializationExcept
 
 
     ClusterProperties property = new ClusterProperties("test");
-    assertEquals(jsonSerializer.fromBytes(jsonSerializer.toBytes(property)), property);
+    ClusterStoreProperties storeProperties = new ClusterStoreProperties(property, null, null);
+    // cluster properties will be serialized then deserialized as cluster store properties
+    assertEquals(jsonSerializer.fromBytes(jsonSerializer.toBytes(property)), storeProperties);
+    // cluster store properties will be serialized then deserialized as cluster store properties
+    assertEquals(jsonSerializer.fromBytes(jsonSerializer.toBytes(storeProperties)), storeProperties);
 
     property = new ClusterProperties("test", schemes);
-    assertEquals(jsonSerializer.fromBytes(jsonSerializer.toBytes(property)), property);
+    assertEquals(jsonSerializer.fromBytes(jsonSerializer.toBytes(property)), new ClusterStoreProperties(property, null, null));
```

Comment:
For backward compatibility the old test of comparing it to the property should still pass too right?

Reply:
nope. As explained in my other comment, The super class's equals method will fail since it sees the class names are not the same (ClusterProperties != ClusterStoreProperties).

Reply:
Backward compatibility is still ensured since the serializer can serialize a ClusterProperties into bytes, and when deserializing (taking bytes back from ZK), it will create a ClusterStoreProperties, which will then be broken down to ClusterProperties again to be saved in LoadBalancerState.

Reply:
We essentially changed the underlining data used in ZK sync (without changing the interface leveraging the superclass-subclass trick), so we bumped the minor version of pegasus to reflect this change.

Reply:
Thanks for the explanation.